FieldsInUse/FieldsInUse/Views/ManageUnusedFieldsDialog.xaml.cs

```diff
@@ -2,37 +2,103 @@
+        private IEnumerable<IRuleRepositoryDefBase> _schemas;
         public ManageUnusedFieldsDialog(RuleApplicationDef ruleAppDef, RuleApplicationController controller)
 	    {
 	        InitializeComponent();
 	        _controller = controller;
 
             var actions = new List<UnusedFieldAction>();
-	        actions.Add(new UnusedFieldAction("Delete", (list, target) =>
-	        {
-	            _controller.RemoveDef(target.InnerDef);
-	            list.RemoveItem(target);
-	        }));
+            _schemas = ruleAppDef.SchemasRoot.AsEnumerable();
+
+            actions.Add(new UnusedFieldAction("Delete", DeleteAction));
             
 	        var columns = FieldDefListView.Options.NameColumn;
 	        columns |= FieldDefListView.Options.TypeNameColumn;
-	        
             
-	        FieldDefListView = new FieldDefListView(ruleAppDef, columns, actions, "Name");
+	        FieldDefListView = new FieldDefListView(ruleAppDef, columns, actions, DeleteManyAction, "Name");
 	        FieldDefListView.OnCloseView += FieldDefListViewOnCloseView;
             
             this.grdMain.Children.Add(FieldDefListView);
 	    }
+        private void DeleteAction(FieldDefListView list, UnusedFieldReference target)
+        {
+            //If the field is bound to a schema, the field must be removed from the schema as well, otherwise there will be validation errors
+            if (target.ExternalSchemaFieldDef != null)
+            {
+                var info = target.ExternalSchemaFieldDef;
+                var parentName = info.AuthoringElementPath.Substring(0, info.AuthoringElementPath.LastIndexOf('.'));
+                var elementName = info.AuthoringElementPath.Replace(parentName, "").Substring(1);
 
-	    private void FieldDefListViewOnCloseView(object sender, object e)
+                var schema = _schemas.FirstOrDefault(s => s.Name == info.AuthoringElementPath.Split('.')[0]);
+                if (schema is XmlSchemaDef xmlSchema)
+                {
+                    if (target.ExternalSchemaFieldDef is EntityDefInfo)
+                    {
+                        var schemaItem = xmlSchema.EntityDefsInfo.EntityDefInfos.FirstOrDefault(i => i.Name == elementName);
+                        xmlSchema.EntityDefsInfo.EntityDefInfos.Remove(schemaItem);
+                    }
+                    else if (target.ExternalSchemaFieldDef is FieldDefInfo)
+                    {
+                        var schemaItem = xmlSchema.EntityDefsInfo.EntityDefInfos.FirstOrDefault(i => i.AuthoringElementPath == parentName);
+                        var fieldItem = schemaItem.Fields[elementName];
+                        schemaItem.Fields.Remove(fieldItem);
+                    }
+                }
+            }
+
+            _controller.RemoveDef(target.InnerDef);
+            list.RemoveItem(target);
+        }
+        public void DeleteManyAction(FieldDefListView list, IEnumerable<UnusedFieldReference> targets)
```

Comment:
for sanity's sake does it make sense to create a temporary backup copy of the ruleapp before making these changes, or at least to prompt the user (if it doesn't already) to save the ruleapp before continuing?